src/api/app/models/scm_webhook.rb

```diff
@@ -112,11 +116,20 @@ def ignored_github_pull_request_action?
     github_pull_request? && ALLOWED_PULL_REQUEST_ACTIONS.exclude?(@payload[:action])
   end
 
+  def ignored_github_push_event?
+    github_push_event? && @payload[:deleted]
+  end
+
   def ignored_gitlab_merge_request_action?
     gitlab_merge_request? && ALLOWED_MERGE_REQUEST_ACTIONS.exclude?(@payload[:action])
   end
 
   def ignored_gitea_pull_request_action?
     gitea_pull_request? && ALLOWED_PULL_REQUEST_ACTIONS.exclude?(@payload[:action])
   end
+
+  def ignored_gitlab_push_event?
+    # In Push Hook events to delete a branch, the after field is '0000000000000000000000000000000000000000'
+    gitlab_push_event? && @payload[:commit_sha] == '0000000000000000000000000000000000000000' # '0' * 40
```

Comment:
```suggestion
    gitlab_push_event? && @payload[:commit_sha].include?('0000000000')
```

Reply:
I think checking with `include?` might be a bit more forgiving

Reply:
it fine as it is

Reply:
I'm going to use a regular expression, wait for it...